podvm-payload/Dockerfile

```diff
@@ -1,6 +1,5 @@
 ## GOLANG ##
-FROM registry.access.redhat.com/ubi9/go-toolset:1.22.9-1739801907 as go_builder
-USER root
+FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_9_golang_1.23 as go_builder
```

Comment:
I guess it might be acceptable to get that as an ARG from the same env file as CAA. WDYT ?

Reply:
No, we can't use arguments for the "FROM" line. Konflux is watching this and updating it for us when needed. It doesn't know variables, and make errors when we do that.
I think Camilla also said the "enterprise contract" requires a clear identification of the builder image, which is not possible when you use variables.
That's why I modified it in the first place. We can't go back on that, we need to keep an actual URL here.

Reply:
Makes me think that I should phrase it out as a comment here maybe?

Reply:
This is unfortunate but I guess we don't have a choice. I think it would make sense to add a comment that this must be hardcoded here. Same with the other impacted dockerfiles.